### ISSUE TITLE
kubelet  parameter(eviction-max-pod-grace-period ), not work as expected like officical comment.

### ISSUE THREAD

**Issue by rainbowBPF2** [kubernetes/kubelet#42]
Hey brothers , feedback a confusing point about kubelet soft eviction:

Just like kubelet --help show: 
--eviction-max-pod-grace-period int32                      Maximum allowed grace period (in seconds) to use when terminating pods in response to a soft eviction threshold being met.  If negative, defer to pod specified value. 

I guess if this parameter is set to negative ,such as -1, soft eviction would use pod specified value(TerminationGracePeriodSeconds),

But When I try to evict pod by creating node pressure(such as memory pressure), I found it's always -1 send to CRI runtime. 
Then pod container stopped immediately with sigkill, exit 137.

In short， this parameter：
（1）set as active number：work as expected；
（2）no set， kubelet read its default number：0， work as expected；
（3）set as negative number: not work as kubelet help show. 

Thanks for your response. May it be a bug ?

**Comment by rainbowBPF2** [comment]
launched again in kubernetes main project ,  ignored here.
